gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/DagProcessingEngineTest.java

```diff
@@ -150,12 +162,19 @@ protected void commit(DagManagementStateStore dagManagementStateStore, Void resu
     }
   }
 
-  // This tests verifies that
+  // This tests verifies that all the dag tasks entered to the dag task stream are retrieved by dag proc engine threads
   @Test
   public void dagProcessingTest() throws InterruptedException, TimeoutException {
+    // there are MAX_NUM_OF_TASKS dag tasks returned and then each thread additionally call (infinitely) once to wait
+    // in this unit tests, it does not infinitely wait though, because the mocked task stream throws an exception on
+    // (MAX_NUM_OF_TASKS + 1) th call
     int expectedNumOfInvocations = MockedDagTaskStream.MAX_NUM_OF_TASKS + ServiceConfigKeys.DEFAULT_NUM_DAG_PROC_THREADS;
+    int expectedExceptions = MockedDagTaskStream.MAX_NUM_OF_TASKS / MockedDagTaskStream.FAILING_DAGS_FREQUENCY;
```

Comment:
this will change back to `MAX_TASKS + numDPEThreads` when the exception is thrown by `DagProc::act`